src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala

```diff
@@ -147,32 +147,94 @@ class WorkflowManagerActor(backend: Backend) extends Actor with CromwellActor {
+      stay()
+    case Event(CallOutputs(workflowId, callName), _) =>
       reply(workflowId, callName, callOutputs(workflowId, callName), WorkflowManagerCallOutputsSuccess, WorkflowManagerCallOutputsFailure)
-    case CallStdoutStderr(workflowId, callName) =>
+      stay()
+    case Event(CallStdoutStderr(workflowId, callName), _) =>
       reply(workflowId, callName, callStdoutStderr(workflowId, callName), WorkflowManagerCallStdoutStderrSuccess, WorkflowManagerCallStdoutStderrFailure)
-    case WorkflowStdoutStderr(id) =>
+      stay()
+    case Event(WorkflowStdoutStderr(id), _) =>
       reply(id, workflowStdoutStderr(id), WorkflowManagerWorkflowStdoutStderrSuccess, WorkflowManagerWorkflowStdoutStderrFailure)
-    case WorkflowMetadata(id) => reply(id, workflowMetadata(id), WorkflowManagerWorkflowMetadataSuccess, WorkflowManagerWorkflowMetadataFailure)
-    case SubscribeToWorkflow(id) =>
+      stay()
+    case Event(WorkflowMetadata(id), _) =>
+      reply(id, workflowMetadata(id), WorkflowManagerWorkflowMetadataSuccess, WorkflowManagerWorkflowMetadataFailure)
+      stay()
+    case Event(SubscribeToWorkflow(id), _) =>
       //  NOTE: This fails silently. Currently we're ok w/ this, but you might not be in the future
       workflowStore.toMap.get(id) foreach {_ ! SubscribeTransitionCallBack(sender())}
-    case RestartWorkflows(w :: ws) =>
+      stay()
+    case Event(RestartWorkflows(w :: ws), _) =>
       restartWorkflow(w)
       context.system.scheduler.scheduleOnce(RestartDelay) {
         self ! RestartWorkflows(ws)
       }
-    case RestartWorkflows(Nil) => // No more workflows need restarting.
-    case CallCaching(id, parameters, callName) =>
+      stay()
+    case Event(RestartWorkflows(Nil), _) =>
+      // No more workflows need restarting.
+      stay()
+    case Event(CallCaching(id, parameters, callName), _) =>
       reply(id, callCaching(id, parameters, callName), WorkflowManagerCallCachingSuccess, WorkflowManagerCallCachingFailure)
+      stay()
+    case Event(Transition(workflowActor, _, toState: WorkflowState), _) if toState.isTerminal =>
+      // Remove terminal actors from the store.
+      removeTerminalWorkflow(workflowActor)
+      stay()
+    case Event(AbortAllWorkflows, _) if workflowStore.isEmpty =>
+      goto(Done)
+    case Event(AbortAllWorkflows, _) =>
+      workflowStore.toMap.values.foreach { _ ! WorkflowActor.AbortWorkflow }
+      goto(Aborting)
+  }
+
+  when (Aborting) {
+    case Event(Transition(workflowActor, _, toState: WorkflowState), _) if toState.isTerminal =>
+      // Remove this terminal actor from the workflowStore and log a progress message.
+      removeTerminalWorkflow(workflowActor)
+      logger.info(s"$tag: Waiting for all workflows to abort (${workflowStore.size} remaining).")
+      // If there are no more workflows to abort we're done, otherwise just stay in the current state.
```

Comment:
The comment is misleading as written, abort messages were already sent to all running workflows.